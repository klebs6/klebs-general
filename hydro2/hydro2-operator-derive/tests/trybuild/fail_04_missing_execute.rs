// ---------------- [ File: hydro2-operator-derive/tests/trybuild/fail_04_missing_execute.rs ]
//! Missing `execute` in the `#[operator(...)]` attribute → compile_fail.
use hydro2_operator_derive::*;
use hydro2_operator::*; 
use named_item_derive::*;
use named_item::*;

#[derive(Debug,NamedItem,Operator)]
#[operator(
    // no "execute=..." → should fail
    opcode="OpCode::TestOp"
)]
pub struct MissingExecuteOp {
    name: String,
}

impl MissingExecuteOp {
    // The macro won't know what function is "the one" to call if we never say `execute="..."`.
    pub async fn some_fn(&self) -> NetResult<()> {
        Ok(())
    }
}

fn main() {}

