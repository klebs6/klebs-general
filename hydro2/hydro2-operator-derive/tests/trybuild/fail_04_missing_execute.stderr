error: No `execute` function name specified in `#[operator(...)]`.
  --> tests/trybuild/fail_04_missing_execute.rs:9:1
   |
9  | / #[operator(
10 | |     // no "execute=..." â†’ should fail
11 | |     opcode="BasicOpCode::TestOp"
12 | | )]
   | |__^
